### ISSUE TITLE
ImageService V2 images.List is broken on 2nd page retrieval

### ISSUE THREAD

**Issue by tatsuhiro-t** [gophercloud/gophercloud#366]
I have a problem getting list of images using ImageService v2 images.List.  First page is OK, but the calculation of the next URL is broken, and it gives me "Resource not found" error.

In particular, in [nextPageURL](https://github.com/gophercloud/gophercloud/blob/d8ccf0f660340eb612d757d632214cc48f87e962/openstack/imageservice/v2/images/urls.go#L41) function, if currentURL is http://example.com/v2/images, and next is /v2/images?marker=..., then the next URL becomes http://example.com/v2/v2/images?marker=....  Notice that extra "/v2" in path component.

**Comment by tatsuhiro-t** [comment]
The following changes work for me:

```go
diff --git a/openstack/imageservice/v2/images/results.go b/openstack/imageservice/v2/images/results.go
index 09996f4..462f28f 100644
--- a/openstack/imageservice/v2/images/results.go
+++ b/openstack/imageservice/v2/images/results.go
@@ -161,7 +161,7 @@ func (r ImagePage) NextPageURL() (string, error) {
                return "", nil
        }
 
-       return nextPageURL(r.URL.String(), s.Next), nil
+       return nextPageURL(r.URL.String(), s.Next)
 }
 
 // ExtractImages interprets the results of a single page from a List() call, producing a slice of Image entities.
diff --git a/openstack/imageservice/v2/images/urls.go b/openstack/imageservice/v2/images/urls.go
index 58cb8f7..bf7cea1 100644
--- a/openstack/imageservice/v2/images/urls.go
+++ b/openstack/imageservice/v2/images/urls.go
@@ -1,7 +1,7 @@
 package images
 
 import (
-       "strings"
+       "net/url"
 
        "github.com/gophercloud/gophercloud"
 )
@@ -38,7 +38,14 @@ func deleteURL(c *gophercloud.ServiceClient, imageID string) string {
 }
 
 // builds next page full url based on current url
-func nextPageURL(currentURL string, next string) string {
-       base := currentURL[:strings.Index(currentURL, "/images")]
-       return base + next
+func nextPageURL(currentURL string, next string) (string, error) {
+       base, err := url.Parse(currentURL)
+       if err != nil {
+               return "", err
+       }
+       rel, err := url.Parse(next)
+       if err != nil {
+               return "", err
+       }
+       return base.ResolveReference(rel).String(), nil
 }
```

**Comment by tatsuhiro-t** [comment]
Fixed by #369 
Thank you for merging the PR.
